src/State/Provider/BackedEnumProvider.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\State\Provider;
+
+use ApiPlatform\Metadata\CollectionOperationInterface;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+
+final class BackedEnumProvider implements ProviderInterface
+{
+    public function __construct(private readonly ProviderInterface $decorated)
+    {
+    }
+
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
+    {
+        $className = $operation->getClass();
+        if (\is_string($className) && enum_exists($className)) {
```

Comment:
return early `$this->decorated->provide($operation, $uriVariables, $context);` ?